integration-tests/integration/calendar.spec.js

```diff
@@ -0,0 +1,54 @@
+const moment = require('moment')
+const CalendarPage = require('../pages/calendarPage')
+const CalendarDetailPage = require('../pages/calendarDetailPage')
+
+context('A staff member can view their calendar', () => {
+  before(() => {
+    cy.task('createTablesInsertData')
+  })
+  beforeEach(() => {
+    cy.task('reset')
+
+    cy.task('stubLogin')
+    cy.task('stubHealthCalls')
+    cy.task('stubStaffLookup')
+    cy.task('stubShifts')
+    cy.login()
+  })
+
+  it('A staff member can view their calendar', () => {
+    const calendarPage = CalendarPage.verifyOnPage(moment().format('MMMM YYYY'))
+
+    // day shift
+    const dayShift = calendarPage.day('2020-03-04')
+    dayShift.should('contain.text', 'Wednesday, 4')
+    dayShift
+      .children('.detail-start')
+      .should('contain.text', 'Start 07:30')
+      .next()
+      .should('contain.text', 'Finish 17:15')
```

Comment:
this is naff.  What I really wanted to do was
```
dayShift.children('.detail-finish').should('contain.text', 'Finish 17:15')
```
however by calling `dayShift.children` the `dayShift` variable then shifts to the child rather than the parent, which is very confusing.

Reply:
any ideas @andrewrlee?

Reply:
I don't think I've come across this before.
We tend to hide the structure of the page within the page object and do separate queries for related elements:
```js
   staffInvolved: (index) => ({
      name: () => cy.get(`#involvedStaff\\[${index}\\]\\[username\\]`),
      remove: () =>
        cy
          .get(`#involvedStaff\\[${index}\\]\\[username\\]`)
          .parents('.add-another__item')
          .find('button.add-another__remove-button'),
    }),
``` 
and then can use it like:
```js
const firstStaff = updatedIncidentDetailsPage.staffInvolved(0)
firstStaff.name().should('have.value', 'AAAA')
firstStaff.remove().should('exist')
```